### PR TITLE
vis: 2016-04-15 -> 2016-07-15

### PR DESCRIPTION
**Pull request by RamKromberg**
###### Motivation for this change

Quite a lot of bug fixes, cleanups and extra features in the last 3 months: https://github.com/martanne/vis/commits/master
###### Things done
- [x] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [x] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/editors/vis/default.nix [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
     repo = "vis";
     owner = "martanne";
   };
 
   nativeBuildInputs = [ makeWrapper ];
 
   buildInputs = [
-     unzip
-     pkgconfig
+     unzip pkgconfig
```
Comment (vrthra) [inline pkgs/applications/editors/vis/default.nix]: Why this change?

File: pkgs/applications/editors/vis/default.nix [hunk 2]
```diff
@@ -33,11 +35,10 @@ stdenv.mkDerivation rec {
...
   '';
 
-  meta = {
+  meta = with stdenv.lib; {
     description = "A vim like editor";
     homepage = http://github.com/martanne/vis;
-    license = stdenv.lib.licenses.isc;
-    maintainers = [ stdenv.lib.maintainers.vrthra ];
+    license = licenses.isc;
+    maintainers = [ maintainers.vrthra ];
```
Comment (vrthra) [inline pkgs/applications/editors/vis/default.nix]: You could add yourself here too, if you are interested in keeping track of Vis

Comment (RamKromberg) [inline pkgs/applications/editors/vis/default.nix]: I usually avoid listing oneself as a maintainer... If I'll find myself updating it again in a couple of month, I'll reconsider :)

**Comment by RamKromberg** [comment]
Quick update: I conditioned the selinux support on linux. Would acl be a problem in other platforms?

File: pkgs/applications/editors/vis/default.nix [hunk 3]
```diff
@@ -1,25 +1,30 @@
-{ stdenv, fetchFromGitHub, unzip, pkgconfig, makeWrapper, ncurses, libtermkey, lpeg, lua }:
+{ stdenv, fetchFromGitHub, unzip, pkgconfig, makeWrapper
+, ncurses, libtermkey, lpeg, lua
+, acl ? null, libselinux ? null
+}:
 
 stdenv.mkDerivation rec {
   name = "vis-nightly-${version}";
-  version = "2016-04-15";
+  version = "2016-07-15";
```
Comment (chexxor) [inline pkgs/applications/editors/vis/default.nix]: I've been thinking lately that moving `version`, `sha256`, and `rev` to the attribute set argument on lines 1-5 is a better place for them. If there, they can easily be overridden in a user's "config.nix" file using the `override` command rather than the `overrideDerivation` command, which is a bit more tricky.

Comment (RamKromberg) [inline pkgs/applications/editors/vis/default.nix]: sounds good to me.

**Comment by RamKromberg** [comment]
Thanks!
